custom_components/tuya_local/devices/madimack_elite_v3_heatpump_updated.yaml

```diff
@@ -0,0 +1,203 @@
+name: Pool heatpump
+products:
+  - id: kwrvh8zwvbbyp086
+    name: Madimack Elite V3 40kW
+primary_entity:
+  entity: climate
+  dps:
+    - id: 101
+      name: hvac_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: auto
+              value: auto
+            - dps_val: cold
+              value: cool
+            - dps_val: heating
+              value: heat
+    - id: 102
+      name: mode
+      type: string
+      hidden: true
+    - id: 104
+      name: temperature
+      type: integer
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              range:
+                min: 54
+                max: 104
+      range:
+        min: 12
+        max: 40
+    - id: 105
+      name: preset_mode
+      type: string
+      translation_key: madimack_presets

```

Comment:
I don't think device specific translation keys are the way to go.  There are currently 627 configs which is a lot of individual icon customisation.

It is better to group into common functionality.  Pool heatpumps fairly commonly support "quiet" "quick" and "smart" presets, though many of them annoyingly have separate presets for cooling and heating modes, and I think we can group these together for both translations and icons.

Reply:
Yep - I gave it a crack to see if it would work on my end and no go. I'll revert the changes